src/lambda-edge/shared/shared.ts

```diff
@@ -447,7 +447,7 @@ function _generateCookieHeaders(
     "spa-auth-edge-nonce-hmac",
     "spa-auth-edge-pkce",
   ].forEach((key) => {
-    cookies[key] = expireCookie();
+    cookies[key] = expireCookie(`;${param.cookieSettings.nonce}`);
```

Comment:
For consistency it might be better to do it exactly the same as a few lines up at L437:

`cookies[key] = expireCookie(cookies[key])`

Will yield the same results, and I guess is more true to your point: the attributes must match those of the cookie that was previously set.

Maybe append an if statement to only do it if the cookie is set to begin with.

What do you think?

Reply:
Now that I look at the few lines up at L437, that L437 may also exhibit the bug.

The execution context at this point is of setting outgoing cookies; _cookies_ is initialized in L381.  The incoming cookies/headers are not readily available at this point in the code.  Also, values of incoming cookies will not have the properties/attributes that were originally set when the cookies were sent out.

L437 may be harder to fix because, within the loop, it will need a mapping between the cookies/keys and the `param.cookieSettings` property to grab the individual cookie settings.

Reply:
Thanks for diving deep!

> The execution context at this point is of setting outgoing cookies; cookies is initialized in L381. The incoming cookies/headers are not readily available at this point in the code. Also, values of incoming cookies will not have the properties/attributes that were originally set when the cookies were sent out.

Ah yes, you're right.

Ok then your implementation is the correct one. Great.